File: PythonAlgorithmInterView/선형자료구조/주식좋은시점.py
def maxProfit(self, prices: list[int]) -> int:
    min_price = 10001
    max_price = 0

    for price in prices :
        min_price = min(min_price, price)
        max_price = max(max_price, price - min_price)

    return max_price



if __name__ == '__main__':
    # [7, 1, 5, 3, 6, 4]
    # prices = [7,6,4,3,1]
    prices = [7, 1, 5, 3, 6, 4]
    maxProfit(None, prices)

File: PythonAlgorithmInterView/선형자료구조/주식좋은시점.js
/**
 * @param {number[]} prices
 * @return {number}
 * 121번 문제.
 */
 var maxProfit = function(prices) {
    let minValue = 10001;
    let maxValue = 0;

    prices.forEach(price=>{
        minValue = Math.min(minValue, price)
        maxValue = Math.max(maxValue, price - minValue)
    })
    return(maxValue)
};


let prices = [7,1,5,3,6,4];
console.log(maxProfit(prices))